bundler/lib/bundler/checksum.rb

```diff
@@ -17,34 +99,23 @@ def initialize_copy(o)
         end
       end
 
-      def [](spec)
-        sums = @store[spec.full_name]
-
-        Checksum.new(spec.name, spec.version, spec.platform, sums&.transform_values(&:digest))
-      end
-
-      def register(spec, checksums)
-        register_full_name(spec.full_name, checksums)
+      def [](full_name)
+        @store[full_name]
       end
 
-      def register_triple(name, version, platform, checksums)
-        register_full_name(GemHelpers.spec_full_name(name, version, platform), checksums)
-      end
-
-      def delete_full_name(full_name)
+      def delete(full_name)
         @store.delete(full_name)
       end
 
-      def register_full_name(full_name, checksums)
-        sums = (@store[full_name] ||= {})
+      def []=(full_name, checksums)
+        delete(full_name)
+        register(full_name, checksums)
+      end
 
-        checksums.each do |checksum|
-          algo = checksum.algo
-          if multi = sums[algo]
-            multi.merge(checksum)
-          else
-            sums[algo] = Multi.new [checksum]
-          end
+      def register(full_name, checksums)
+        Array(checksums).each do |checksum|
+          @store[full_name][checksum.algo] ||= checksum
+          @store[full_name][checksum.algo].merge(checksum)
```

Comment:
LOL. Atrocious but no extra merge. (Sorry)

```suggestion
          @store[full_name][checksum.algo]&.merge(checksum)
          @store[full_name][checksum.algo] ||= checksum

```